src/.clang-tidy

```diff
@@ -0,0 +1,298 @@
+---
```

Comment:
Should this hidden file be checked in?

Reply:
clang-tidy behavior is configured and customized using .clang-tidy file(s) in source directory. I updated the comment in the file to describe how we can generate it and tweak it.